src/server/utils/validation/validators.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2017-2022 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Timezone } from "chronoshift";
+import { Request } from "express";
+import { Expression } from "plywood";
+import { deserialize } from "../../../client/deserializers/app-settings";
+import { AppSettings, ClientAppSettings, serialize } from "../../../common/models/app-settings/app-settings";
+import { ClientDataCube } from "../../../common/models/data-cube/data-cube";
+import { isQueryable, QueryableDataCube } from "../../../common/models/data-cube/queryable-data-cube";
+import { Essence } from "../../../common/models/essence/essence";
+import { getDataCube, Sources } from "../../../common/models/sources/sources";
+import { isNil } from "../../../common/utils/general/general";
+import { definitionConverters, ViewDefinition, ViewDefinitionVersion } from "../../../common/view-definitions";
+import { ViewDefinitionConverter } from "../../../common/view-definitions/view-definition-converter";
+import { checkAccess } from "../datacube-guard/datacube-guard";
+import { SettingsManager } from "../settings-manager/settings-manager";
+
+export class ValidationError extends Error {
```

Comment:
It's rather InvalidRequestError (as long as status code is 400)

Reply:
We can have different codes here.